packages/block-directory/package.json

```diff
@@ -47,8 +47,8 @@
 		"change-case": "^4.1.2"
 	},
 	"peerDependencies": {
-		"react": "^17.0.0",
-		"react-dom": "^17.0.0"
+		"react": "^18.0.0",
```

Comment:
In order for the rest of the ecosystem to update smoothly and at their own pace, should we support both 17 and 18 as peer deps in most packages? (As long as our code is still compatible with 17, that is)

Reply:
I'm not sure if our code will be compatible, that's the problem.

For example, I suspect that we'll need to update the `useSelect` hook to use `useSyncExternalStore` from React 18, otherwise it will not work 100% reliably with v18. Then the `@wordpress/data` library has a non-negotiable dependency on React 18, and that transitively affects all other libraries...

Let's keep the v17 compat requirement in mind, and see what we can do. Libraries like `react-redux` use a compat shim for `useSyncExternalStore`, and that allows them to support older versions back to React 16.